Documentation/ApiOverview/Events/Events/Linkvalidator/BeforeRecordIsAnalyzedEvent.rst

```diff
@@ -10,6 +10,144 @@ BeforeRecordIsAnalyzedEvent
+:php:`TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent`.  By default
+the method is called :php:`handleEvent`:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace Vendor\MyExtension\EventListener\Linkvalidator;
+
+    use TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent;
+
+    class BeforeRecordIsAnalyzedEventListener
+    {
+
+        public function handleEvent(BeforeRecordIsAnalyzedEvent $event): void
+        {
+            // do something
+        }
+    }
+
+The listener must then be registered in the extensions :php:`Services.yaml`:
+
+..  code-block:: yaml
+    :caption: EXT:my_extension/Configuration/Services.yaml
+
+    services:
+        _defaults:
+            autowire: true
+            autoconfigure: true
+            public: false
+
+        Vendor\MyExtension\:
+            resource: '../Classes/*'
+            exclude: '../Classes/Domain/Model/*'
+
+        Vendor\MyExtension\EventListener\Linkvalidator\BeforeRecordIsAnalyzedEventListener:
+            tags:
+                - name: event.listener
+                  method: handleEvent
+                  identifier: 'myExtensionBeforeRecordIsAnalyzedEventListener'
+                  event: TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent
+
+For the implementation we need the :php:`BrokenLinkRepository` to register
+additional link errors and the :php:`SoftReferenceParserFactory` so we can
+automatically parse for links. These two classes have to be injected via
+:ref:`dependeny injection <Dependency-Injection>`:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    use TYPO3\CMS\Core\DataHandling\SoftReference\SoftReferenceParserFactory;
+    use TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent;
+    use TYPO3\CMS\Linkvalidator\Repository\BrokenLinkRepository;
+
+    class BeforeRecordIsAnalyzedEventListener
+    {
+        protected BrokenLinkRepository $brokenLinkRepository;
+        protected SoftReferenceParserFactory $softReferenceParserFactory;
+
+        public function __construct(
+            BrokenLinkRepository $brokenLinkRepository,
+            SoftReferenceParserFactory $softReferenceParserFactory
+        ) {
+            $this->brokenLinkRepository = $brokenLinkRepository;
+            $this->softReferenceParserFactory = $softReferenceParserFactory;
+        }
+    }
+
+In this example we only check the field :sql:`bodytext` of new tables:
+
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php
+
+    public function handleEvent(BeforeRecordIsAnalyzedEvent $event): void
+    {
+        $table = $event->getTableName();
+        $forbiddenDomain = 'example.org';
+        if ($table == 'tx_news_domain_model_news') {
```

Comment:
Suggestion, use negative condition (`$table !== ...`) and return early. This avoids at least on indentation. Also use indentical check, not equal check as best practise.